test/e2e/node_feature_discovery_test.go

```diff
@@ -772,6 +772,57 @@ core:
 				})
 			})
 		})
+
+		Context("Deploy single nfd-master pod with config file", Ordered, func() {
+			BeforeEach(func() {
+				err := testutils.ConfigureRBAC(f.ClientSet, f.Namespace.Name)
+				Expect(err).NotTo(HaveOccurred())
+
+			})
+
+			AfterEach(func() {
+				Expect(testutils.DeconfigureRBAC(f.ClientSet, f.Namespace.Name)).NotTo(HaveOccurred())
+			})
+
+			It("Should deploy nfd-master pod with config map successfully", func() {
+				By("Creating nfd-master config")
+				cm := testutils.NewConfigMap("nfd-master-conf", "nfd-master.conf", `
+enableTaints: true
+`)
+				cm, err := f.ClientSet.CoreV1().ConfigMaps(f.Namespace.Name).Create(context.TODO(), cm, metav1.CreateOptions{})
+				Expect(err).NotTo(HaveOccurred())
+
+				// Launch nfd-master
+				By("Creating nfd-master pod")
+				podSpecOpts := createPodSpecOpts(
+					testpod.SpecWithContainerImage(dockerImage()),
+					testpod.SpecWithConfigMap(cm.Name, "/etc/kubernetes/node-feature-discovery"),
+					testpod.SpecWithTolerations(testTolerations),
+				)
+				masterPod := e2epod.NewPodClient(f).CreateSync(testpod.NFDMaster(podSpecOpts...))
+
+				By("Waiting for the nfd-master pod to be running")
+				Expect(e2epod.WaitTimeoutForPodRunningInNamespace(f.ClientSet, masterPod.Name, masterPod.Namespace, time.Minute)).NotTo(HaveOccurred())
+
+				By("Verifying the node where nfd-master is running")
+				// Get updated masterPod object (we want to know where it was scheduled)
+				masterPod, err = f.ClientSet.CoreV1().Pods(f.Namespace.Name).Get(context.TODO(), masterPod.Name, metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				// Node running nfd-master should have master version annotation
+				masterPodNode, err := f.ClientSet.CoreV1().Nodes().Get(context.TODO(), masterPod.Spec.NodeName, metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				Expect(masterPodNode.Annotations).To(HaveKey(nfdv1alpha1.AnnotationNs + "/master.version"))
```

Comment:
I think we'd like to verify that the config file was actually taken into effect by nfd-master. We don't necessarily need to verify the annotations here as it's done already in other tests (just to speed up things a bit).

What we could do, for example
1. deploy a NodeFeature object to create some features on one worker node
2. deploy NodeFeatureRule to taint the node
3. Verify that the expected taint was created
4. Modify the configmap to `enableTaints: false``
5. Verify that the taint was removed from the node

For the NodeFeature object you could probably just re-use `test/e2e/data/nodefeature-1.yaml`. For the NodeFeatureRule you could create something new, simple like
```yaml
spec:
  rules:
    - name: "e2e-tainting-test"
      taints:
        - effect: PreferNoSchedule
          key: "nfd.node.kubernetes.io/e2e-test"
      matchFeatures:
        - feature: "e2e.flags"
          matchExpressions:
            "flag_1": {op: Exists}
```

```suggestion
```

Reply:
You can take a look at other tests, e.g. those for taints for some inspiration how the taints are created and verified